abide/src/main/scala/scala/tools/abide/traversal/ScopingRule.scala

```diff
@@ -1,40 +1,1113 @@
+      if (context.inBlock)
+        context.scope enter sym
+
+      val defDefContext = context.makeNewScope(ddef, sym)
+      val isConstrDefaultGetter = ddef.mods.hasDefault && sym.owner.isModuleClass &&
+        nme.defaultGetterToMethod(sym.name) == nme.CONSTRUCTOR
+      val newContext = if (isConstrDefaultGetter) defDefContext.makeConstructorContext else defDefContext
+
+      newContext.scope enter sym
+      for (tparam <- ddef.tparams) newContext.scope enter tparam.symbol
+      for (vparams <- ddef.vparamss; vparam <- vparams) newContext.scope enter vparam.symbol
+
+      withContext(newContext)
+    }
+
+    private[ScopingRuleTraversal] def enterTypeDef(tdef: TypeDef): State = {
+      val sym = tdef.symbol
+      if (context.inBlock)
+        context.scope enter sym
+
+      val newContext = context.makeNewScope(tdef, sym)
+      for (tparam <- tdef.tparams) newContext.scope enter tparam.symbol
+      withContext(newContext)
+    }
+
+    private[ScopingRuleTraversal] def enterFunction(fun: Function): State = {
+      val newContext = context.makeNewScope(fun, fun.symbol)
+      for (vparam <- fun.vparams) newContext.scope enter vparam.symbol
+      withContext(newContext)
+    }
+
+    private[ScopingRuleTraversal] def enterClassDef(cd: ClassDef): State = {
+      val newContext = context.makeNewScope(cd, cd.symbol)
+      for (tparam <- cd.tparams) newContext.scope enter tparam.symbol
+      withContext(context.makeNewScope(cd, cd.symbol))
+    }
+
+    private[ScopingRuleTraversal] def enterModuleDef(md: ModuleDef): State =
+      withContext(context.makeNewScope(md, md.symbol.moduleClass))
+
+    private[ScopingRuleTraversal] def enterPackageDef(pd: PackageDef): State =
+      withContext(context.make(pd, pd.symbol.moduleClass, pd.symbol.info.decls))
+
+    private[ScopingRuleTraversal] def enterTemplate(template: Template): State = {
+      val ctx = context.make(template, context.owner, newScope)
+      if (template.self.name != nme.WILDCARD) ctx.scope enter template.self.symbol
+      withContext(ctx)
+    }
+
+    private[ScopingRuleTraversal] def enterBlock(block: Block): State =
+      withContext(context.makeNewScope(block, context.owner))
+
+    private[ScopingRuleTraversal] def enterImport(i: Import): State =
+      withContext(context.make(i))
+
+    private[ScopingRuleTraversal] def enterCaseDef(cd: CaseDef): State =
+      withContext(context.makeNewScope(cd, context.owner))
+
+    private[ScopingRuleTraversal] def enterBind(b: Bind): State = {
+      context.scope enter b.symbol
+      withContext(context)
+    }
+
+    private[ScopingRuleTraversal] def enterApply(a: Apply): State =
+      withContext(context.make(a))
+
+    private[ScopingRuleTraversal] def leaveContext: State = withContext(context.parent)
+
+    /** Helper method to extract symbol from context given name and filter */
+    def lookup(name: Name, qualifies: Symbol => Boolean): NameLookup = context.lookupSymbol(name, qualifies)
```

Comment:
I know it's a nitpick, but I'd like to rename `qualified` to one of `suchThat` and `p`. One is pretty standard in scalac parlance, the other one is what `filter` and others are using to designate a predicate.